adapters/coap/src/test/java/org/eclipse/hono/adapter/coap/CoapContextTest.java

```diff
@@ -162,4 +167,76 @@ void testStartAckTimerHandlesNonNumberPropertyValue() {
         ctx.startAcceptTimer(vertx, tenant, 500);
         verify(vertx).setTimer(eq(500L), VertxMockSupport.anyHandler());
     }
+
+    /**
+     * Verifies that the CoAP time option is not in the response if not requested by either a request option or parameter.
+     */
+    @Test
+    void testTimeOptionIsNotIncludedInResponseIfNotRequested() {
+        final CoapExchange exchange = mock(CoapExchange.class);
+        final Adapter coapConfig = new Adapter(Constants.PROTOCOL_ADAPTER_TYPE_COAP);
+        final TenantObject tenant = TenantObject.from("tenant", true).addAdapter(coapConfig);
+        final var authenticatedDevice = new DeviceUser(tenant.getTenantId(), "device-id");
+        final CoapContext ctx = CoapContext.fromRequest(exchange, authenticatedDevice, authenticatedDevice, "4711", span);
+        final Response response = mock(Response.class);
+        final OptionSet responseOptions = new OptionSet();
+        when(response.getOptions()).thenReturn(responseOptions);
+        ctx.respond(response);
+        assertThat(responseOptions.hasOption(TimeOption.COAP_OPTION_TIME_NUMBER)).isFalse();
+    }
+
+
+    /**
+     * Ensures that a timestamp happened within a half second ago.
+     *
+     * @param timestamp the timestamp to check
+     */
+    private void assertTimestampWithinRangeOfCurrentTime(final long timestamp) {
```

Comment:
I wonder what the purpose of this assertion might be? FMPOV the adapter will not be able to guarantee that the response message takes less than 500ms to reach the client, can it?